app/models/order_item.rb

```diff
@@ -1,16 +1,31 @@
 class OrderItem < ActiveRecord::Base
   belongs_to :book
   belongs_to :order
+  belongs_to :cart
 
   before_save :set_unit_price_and_total
 
   validates :quantity, presence: true, numericality: { greater_than: 0 }
-  validates :book, :order, presence: true
+  validates :book, presence: true
+
+  delegate :title, to: :book
+
+  def unit_price
+    if persisted?
+      self[:unit_price]
+    else
+      book.price
+    end
+  end
+
+  def total_price
+    unit_price * quantity
+  end
 
   private
 
   def set_unit_price_and_total
```

Comment:
Is this total price behavior a result of the same thing? Seems like something that should be dynamically calculated.